tests/api_workflow/test_api_workflow_upload_custom_metadata.py

```diff
@@ -0,0 +1,71 @@
+import copy
+import json
+import os
+import random
+import tempfile
+import pathlib
+
+import numpy as np
+import torchvision
+
+from lightly.api.utils import MAXIMUM_FILENAME_LENGTH
+from lightly.data.dataset import LightlyDataset
+
+from tests.api_workflow.mocked_api_workflow_client import \
+    MockedApiWorkflowSetup
+
+import cv2
+
+
+class TestApiWorkflowUploadCustomMetadata(MockedApiWorkflowSetup):
+
+    def create_fake_dataset(self, n_data: int = 10, sample_names=None):
+        self.dataset = torchvision.datasets.FakeData(size=n_data,
+                                                     image_size=(3, 32, 32))
+
+        self.folder_path = tempfile.mkdtemp()
+        image_extension = '.jpg'
+        sample_names = sample_names if sample_names is not None else [
+            f'img_{i}{image_extension}' for i in range(n_data)]
+        for sample_idx in range(n_data):
+            data = self.dataset[sample_idx]
+            sample_name = sample_names[sample_idx]
+            path = os.path.join(self.folder_path, sample_name)
+            data[0].save(path)
+
+        coco_json = dict()
+        coco_json['images'] = [{'id': i, 'file_name': fname} for i, fname in
+                               enumerate(sample_names)]
+        coco_json['metadata'] = [{'id': i, 'image_id': i, 'custom_metadata': 0}
+                                 for i, _ in enumerate(sample_names)]
```

Comment:
Maybe use `COCO_ANNOTATION_KEYS.images` instead of `'images'`  for consistency. Same for `'metadata'`